webapp/channels/src/components/suggestion/switch_channel_provider.tsx

```diff
@@ -421,7 +456,7 @@ export default class SwitchChannelProvider extends Provider {
 
             // Dispatch suggestions for local data (filter out deleted and archived channels from local store data)
             const channels = getChannelsInAllTeams(getState()).concat(getDirectAndGroupChannels(getState())).filter((c) => c.delete_at === 0);
-            const users = Object.assign([], searchProfilesMatchingWithTerm(getState(), channelPrefix, false));
+            const users = searchProfilesMatchingWithTerm(getState(), channelPrefix, false);
```

Comment:
There's quite a few spots in this file where I stopped us from copying objects randomly. I think a lot of it predates back to pre-Redux days where we had to be a lot more careful about not modifying data that came from the store